### ISSUE TITLE
Nvidia 1050 TI GPU Compatibility issues

### ISSUE THREAD

**Issue by Wheeler-Colton** [DrewThomasson/ebook2audiobook#2040]
Hello! I have been trying for about 6 hours trying to get this running in a docker compose environment. I have a 1050 ti, and have tried all the combinations that i can think of. I have followed this guide:

https://github.com/DrewThomasson/ebook2audiobook/wiki/GPU-ISSUES
And running the build command with the DEVICE_TAG=124 and DEVICE_TAG=118. 

(Side note, this documentation doesn't work:
        DEVICE_TAG=[XXX] ./ebook2audiobook.sh --script_mode build_docker --script_mode compose
you need to run it with the docker mode option instead of script mode)

I have tried doing the device tag, after the build with this command:

DEVICE_TAG=cu118 / 124 docker compose --profile cuda up --no-log-prefix

I am attempting 128 now. That is what the log file will be attached. 

Reading the logs after my first attempt, It recognizes my GPU, recognizes my file and then it starts to process.. but fails.  it states i should download a version of PyTorch that supports 12.6, but if i put that in the DEVICE_TAG=cu126 it downloads the latest version which utilizes cu130. At least as far as i can understand. Additionally it also states my GPU has compute capabilities of 6.1. 

It then gives me a list. Is there a way to lock that version in to attempt again? 

I am using the predefined docker compose. 

Any help or advice is appreciated. I also understand if my GPU is not compatible.

1)**Operating System:**
 - Ubuntu 24.04 (linux)

2)**Hardware and Model**
 - Cuda 12.6 (From what i understand that is what my GPU maxes out at) 
 - Nvidia 1050 ti

3)**Script Mode**
- Docker Compose
 
4)**Full Terminal log As Attached File**
Attempt Logs: 
[_ebook2audiobook_logs.txt](https://github.com/user-attachments/files/30429068/_ebook2audiobook_logs.txt)

**Comment by ROBERT-MCDOWELL** [comment]
> (Side note, this documentation doesn't work:
> DEVICE_TAG=[XXX] ./ebook2audiobook.sh --script_mode build_docker --script_mode compose
> you need to run it with the docker mode option instead of script mode)

fixed thanks.

from your log, you are using e2a v26.4.28 which is old already. First clone the very last git repo or download the v26.7.23
then just launch:
./e2a.sh --script_mode build_docker --docker_mode compose

after that, thanks to provide the build log and what's inside .device_info.json file

**Comment by Wheeler-Colton** [comment]
> from your log, you are using e2a v26.4.28 which is old already. First clone the very last git repo or download the v26.7.23
> then just launch:

I think this ended up being that way because i did the device_tag=cu128 which has a repository on docker hub, with that old repository. 

I completely removed, recloned, and rebuilt to make sure nothing was done incorrectly. 

Here is the Build Log File:

[e2a_BuildLogs.txt](https://github.com/user-attachments/files/30434663/e2a_BuildLogs.txt)

Here is the .device_info.json

`{"name": "cuda", "os": "manylinux_2_28", "arch": "x86_64", "pyvenv": [3, 12], "tag": "cu124", "note": ""}`

**Comment by ROBERT-MCDOWELL** [comment]
everything looks good. do you have CUDA runtime 12.4 installed? it should be, as e2a detected your card works with cuda 12.4 and torch 2.6..0 only.

**Comment by ROBERT-MCDOWELL** [comment]
try to convert an ebook and tell me if it works.

**Comment by Wheeler-Colton** [comment]
Ran this:

`DEVICE_TAG=cu124 docker compose --profile cuda up --no-log-prefix`

[+] up 1/1
 ✔ Container ebook2audiobook-ebook2audiobook-cuda-1 Created                                                                                                                   0.0s
Attaching to ebook2audiobook-cuda-1
v26.7.23 full_docker mode


This is what i get when i attempt to do a book. 

[_ebook2audiobook-ebook2audiobook-cuda-1_logs.txt](https://github.com/user-attachments/files/30436657/_ebook2audiobook-ebook2audiobook-cuda-1_logs.txt)

**Comment by ROBERT-MCDOWELL** [comment]
interesting. A package seems to override the right torch/torchaudio version. let me check.

**Comment by ROBERT-MCDOWELL** [comment]
ok found it. it seems the pkg pydiantic changed his behavior and does not see torch 2.6.0 for whatever reason, then it installs the last torch.

**Comment by ROBERT-MCDOWELL** [comment]
could you try to install it in native to see if the installation gives the same bug?

**Comment by Wheeler-Colton** [comment]
Extremely long as i usually keep my host pretty clean. 

I am getting a different error, but it at least seems to be trying to process. 

[Native_e2a_install_run_logs.txt](https://github.com/user-attachments/files/30437741/Native_e2a_install_run_logs.txt)

The very bottom is where i ran it. Once on Tortoise, thought i overloaded the GPU, then on Piper. I may have missed something in the installation, of the File may be no good. It is an epub i have had for quite some time. 


I did CHMOD the directory with full permissions to fix those issues for testing purposes.

**Comment by ROBERT-MCDOWELL** [comment]
> Extremely long as i usually keep my host pretty clean.

no worries, you will just run ./uninstall.sh to clean back your host.


> I did CHMOD the directory with full permissions to fix those issues for testing purposes.

yes you need the same user when installing and running gradio. if not, chmod 777 the folders where gradio is writing like audiobooks, tmp, run, models.

so native works for you right?

**Comment by Wheeler-Colton** [comment]
No, sorry. I had errors processing the files still. I did chmod 777

They were just different errors.

**Comment by Wheeler-Colton** [comment]
Cleaned up log test file. 

First Tortoise then Piper.

[Clean_Native.txt](https://github.com/user-attachments/files/30438238/Clean_Native.txt)

**Comment by ROBERT-MCDOWELL** [comment]
ok after read your native log I confirm it's the same issue although it's a different error log.
your GPU being between 2 radical configuration involving several pkg creating dependencies confusion, apparently
I need to force any gpu requesting cu124 to cu126 as torch last version supporting it and btw capabilities 6.1

**Comment by ROBERT-MCDOWELL** [comment]
could you try before to uninstall the native version:

```
conda activate ./python_env
python -c "
import torch, torchaudio, torchaudio.transforms as T
print(torch.__version__, torchaudio.__version__)
print(T.Resample(44100, 16000)(torch.randn(1, 44100)).shape)
print(T.MelSpectrogram()(torch.randn(1, 16000)).shape)
"
conda deaactivate
```

**Comment by Wheeler-Colton** [comment]
Is this what you need?

[conda.txt](https://github.com/user-attachments/files/30439126/conda.txt)

**Comment by ROBERT-MCDOWELL** [comment]
yes thanks. I will fix it asap. I just need you test it again with my coming patches.

**Comment by Wheeler-Colton** [comment]
Appreciate the application! Just let me know and i will be happy to test. Are you okay if i run the Uninstall command now?

**Comment by ROBERT-MCDOWELL** [comment]
yes you can, it will be better to reinstall the patched version cleanly.

**Comment by ROBERT-MCDOWELL** [comment]
ok you can git pull my fork and install again in native mode. if it works try to build the docker image it should work also.
https://github.com/ROBERT-MCDOWELL/ebook2audiobook

**Comment by Wheeler-Colton** [comment]
Alright. The native build seemed to work. 

Here is the build: 

[Fork_Native_Install.txt](https://github.com/user-attachments/files/30473447/Fork_Native_Install.txt)

The only thing is Tortoise apparently is too large for my CUDA memory. It kept failing. Trying to switch to piper, i get similar issues. 

I ended up getting it to work on vits.

Here is the completed log of the conversion. (Messy as i was troubleshooting in the middle of it. 1 Tortoise, 1 piper, 2 - 3 Tortoise, then 1 vits)

[New_Native_Conversion.txt](https://github.com/user-attachments/files/30473639/New_Native_Conversion.txt)

**Comment by ROBERT-MCDOWELL** [comment]
another issue with onnxruntime. causing also the false positive of memory full.
make a test and return the result:
```
conda activate ./python_env
python -c "import onnxruntime as ort; print(ort.__version__, ort.get_available_providers())"
conda deactivate
```

**Comment by ROBERT-MCDOWELL** [comment]
in the same time just git pull my fork again and re-run.

**Comment by ROBERT-MCDOWELL** [comment]
STOP! please keep my fork intact. compress the folder and share it to  me, something wrong happened with some git commands thanks!

**Comment by Wheeler-Colton** [comment]
Sorry was i messing with your fork? I wasn't trying to. I was just waiting for the Audio to process. Then trying to give you the logs of conda. 

[Conda_fork.txt](https://github.com/user-attachments/files/30510964/Conda_fork.txt)

What folder do you need me to compress? 

I wasn't trying to push anything, so what happened so i can avoid it next time?

**Comment by ROBERT-MCDOWELL** [comment]
ok I just fixed my fork. you can clone it again sorry.

**Comment by ROBERT-MCDOWELL** [comment]
but before to do anything, if you still have my fork installed. just do

```
conda activate ./python_env
python -c "import onnxruntime as ort; print(ort.__version__, ort.get_available_providers())"
conda deactivate
```

**Comment by ROBERT-MCDOWELL** [comment]
I see you already done it. but did you do it before to re-clone my fork or after?

**Comment by ROBERT-MCDOWELL** [comment]
ok now run:

```
conda activate ./python_env
python -c "import onnxruntime as ort; print(ort.__file__, ort.__path__)"
pip list 2>/dev/null | grep -i onnxruntime
ls python_env/lib/python3.12/site-packages/onnxruntime/
```

if it's return None:

```
pip uninstall -y onnxruntime onnxruntime-gpu onnxruntime-directml
rm -rf python_env/lib/python3.12/site-packages/onnxruntime
pip install "onnxruntime-gpu<1.27"

python -c "import onnxruntime as ort; print(ort.__version__, ort.get_device(), ort.get_available_providers())"

python -c "
import onnxruntime as ort, glob
model = glob.glob('models/tts/**/*.onnx', recursive=True)[0]
s = ort.InferenceSession(model, providers=['CUDAExecutionProvider'])
print(model, s.get_providers())
"
```

**Comment by Wheeler-Colton** [comment]
I did it before recloning your fork. I was just getting to it this morning, opened to send it and saw your messages. 

Do you want me to reclone and then run those commands?

**Comment by Wheeler-Colton** [comment]
Before Recloning: 

[Yesterdays_Fork_ONNX.txt](https://github.com/user-attachments/files/30511981/Yesterdays_Fork_ONNX.txt)

**Comment by ROBERT-MCDOWELL** [comment]
> I did it before recloning your fork. I was just getting to it this morning, opened to send it and saw your messages.
> 
> Do you want me to reclone and then run those commands?

yes please

**Comment by Wheeler-Colton** [comment]
Recloned Native Install File Conversion test.

[Conversion.log](https://github.com/user-attachments/files/30522203/Conversion.log)

Tortoise is still causing Issues. but xtts worked great. 

Conda ONNX:

[ReClone_Conda_ONNXRuntime.txt](https://github.com/user-attachments/files/30522108/ReClone_Conda_ONNXRuntime.txt)

Since it didn't say none, I didn't run the Second Command. 

Let me know if you need me to run that.

**Comment by ROBERT-MCDOWELL** [comment]
I have the right info thanks.
what is your CUDA runtime version installed?

**Comment by ROBERT-MCDOWELL** [comment]
could you git pull from my fork and run again ./e2a.sh?
it should downgrade onnxruntime-gpu < 1.27.0 and work for all TTS.
you can test the conversion with tortoise, xtts, piper at least.

**Comment by Wheeler-Colton** [comment]
> what is your CUDA runtime version installed?

nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2024 NVIDIA Corporation
Built on Thu_Mar_28_02:18:24_PDT_2024
Cuda compilation tools, release 12.4, V12.4.131
Build cuda_12.4.r12.4/compiler.34097967_0

e2a.sh Build:

[e2a_sh_build.txt](https://github.com/user-attachments/files/30551226/e2a_sh_build.txt)



Models Tests:

[Tortoise.log](https://github.com/user-attachments/files/30551446/Tortoise.log)

(Failed with still an out of Memory issue)

Piper:

[Piper.log](https://github.com/user-attachments/files/30551476/Piper.log)

(Worked)

XTTS: 

[XTTS.log](https://github.com/user-attachments/files/30551767/XTTS.log)

(Worked)

______________
Honestly happy with the performance of XTTS. I don't need Tortoise to work. 

Are we at a point we can test this in a docker compose environment now?

**Comment by ROBERT-MCDOWELL** [comment]
ok fine, tortoise seems to ask a little more VRAM to work. so I should increase to 6GB min. if not raise a warning that it can crash.
but before I updated its class so it would be nice if you  could "git pull" and test the conversion again.
after that. if you use docker compose I must PR to the main branch and the workflow will update the docker image. you could use docker rather than compose to build locally the image with ./e2a.sh --script_mode build_docker without --docker_mode.

**Comment by Wheeler-Colton** [comment]
Ultimately i would like to get it into a compose environment, as i prefer to run everything that way so it is easy to modify some of my settings IE: Traefik Labels, Loki Logging, Diun. 



> but before I updated its class so it would be nice if you could "git pull" and test the conversion again.

Which conversion do you want me to test? Any of them? Or a specific one?

**Comment by ROBERT-MCDOWELL** [comment]
ok so for compose you need to wait until the PR is merged to the main here. I should do it today or tomorrow.
only tortoise conversion would be fine thanks.

**Comment by Wheeler-Colton** [comment]
Failed instantly with a note saying i didn't have enough GB.

[Tortoise.log](https://github.com/user-attachments/files/30555685/Tortoise.log)

I will wait for the compose tomorrow. then i will try it again. 

Thanks for all your help.

**Comment by ROBERT-MCDOWELL** [comment]
ok perfect. this is the right behavior now. tortoise needs more than 6GB VRAM or just use your CPU but it's slow.
the docker image should have the same results for all TTS btw.

**Comment by ROBERT-MCDOWELL** [comment]
you can test tortoise a last time after a "git pull", I think I made it work for old gpu with limited VRAM.

**Comment by ROBERT-MCDOWELL** [comment]
docker is updated, but check the version on gradio to be sure.

**Comment by Wheeler-Colton** [comment]
Docker is working great. My Custom compose is as well. I am using XTTS. 

Tortoise still failed, but i think it is because i didn't end up pulling again. I uninstalled yesterday in preparation for the Docker.

**Comment by ROBERT-MCDOWELL** [comment]
yes tortoise really needs 6GB VRAM and fails. but if the CPU / RAM is used it works now even if it's under 6GB but more slower than XTTSv2 which is better quality.

**Comment by ROBERT-MCDOWELL** [comment]
closing for now. thanks for your tests!
